fullstop-whitelist/README.md

```diff
@@ -0,0 +1,102 @@
+exact expiration date is not yet known, a rule can also be invalidated at any later point in time. This is the only
+possible modification to an existing rule. Updating the checks of a rule, will result in the old one being invalidated
+and a new rule with updated check criteria being created. This way, traceability of whitelisting decisions is guaranteed.
+
+## Whitelisting criteria
+
+One rule can have multiple criteria. Only if all of them are met, the rule is applied successfully.
+
+### account_id
+ID (12-digit) of an AWS account.
+Matches if the violation occurred in this AWS account.
+
+Example: `"111222333444"` 
+
+### application_id
+ID of an application, as known from the Kio application registry (usually only lower case chars).
+Matches if the violation could be assigned to this application.
+
+Example: `"pierone"`
+
+### application_version
+Version of an application, as specified in the Taupage YAML.
+Matches if the violation could be assigned to this application version. 
+
+Example: `"v1"`
+
+### image_name
+Regular expression.
+Matches if the Amazon Machine Image (AMI) name matches the regex.
+
+Example: `"^CoreOS-stable-[0-9.]+-hvm$"`
+
+### image_owner
+ID (12-digit) of an AWS account.
+Matches if the owner of the AMI is exactly this AWS account.
+
+Example: `"111222333444"`
+
+### meta_info_json_path
+[JsonPath](https://github.com/json-path/JsonPath) expression, which should return a list.
+Matches if, when applied to the the meta_info property (which is an arbitrary JSON object),
+returns a non-empty list of results. JsonPath allows for complex queries.
+
+Example: `"$.[?(@.role_name == 'MyReadOnlyRole' && @.grantees == ['arn:aws:iam::999888777666:root'])]"`
+
+When this expression is applied to the sample meta info:
+```json
+{
+  "role_name": "MyReadOnlyRole",
+  "grantees": [
+    "999888777666"
+  ],
+  "role_arn": "arn:aws:iam::111222333444:role/MyReadOnlyRole"
+}
+```
+it will return the result:
+```json
+[
+   {
+      "role_name" : "MyReadOnlyRole",
+      "grantees" : [
+         "arn:aws:iam::999888777666:root"
+      ],
+      "role_arn" : "arn:aws:iam::111222333444:role/MyReadOnlyRole"
+   }
+]
+```
+When we change the expression slightly, e.g. to match for another role name:
+`"$.[?(@.role_name == 'AdminRole' && @.grantees == ['arn:aws:iam::999888777666:root'])]"`
+and apply to the sample meta info, it will return an empty list `[]`.
+
+**Hint:** Use the [Jayway JsonPath Evaluator](http://jsonpath.herokuapp.com/)
+to try out the expressions before using them in a rule.
+
+### region
+ID of an AWS region (see [AWS docs](http://docs.aws.amazon.com/general/latest/gr/rande.html) for a complete list).
+Matches if the violation occurred in this region.
+
+Example: `"eu-central-1"`
+
+### violation_type
+ID of the violation type (use command line tool `fullstop types` to get a complete list).
+Matches if the violation was of exactly this type.
+
+Example: `"UNSECURED_PUBLIC_ENDPOINT"`
```

Comment:
We could add a complete json example of a whitelisting rule like this:

{
  "account_id": "123456789",
  "reason": "Aligned with Management. Team is allowed to use other AMI's until the end of the year.",
  "violation_type": "WRONG_AMI",
  "expiry_date": "2017-12-31T00:00:00.000Z"
}

Reply:
I added a few complete examples a the bottom of the page